cpp/benchmarks/join/generate_input_tables.cuh

```diff
@@ -45,13 +46,14 @@ __global__ static void init_build_tbl(key_type* const build_tbl,
                                       curandState* state,
                                       int const num_states)
 {
-  auto const start_idx = blockIdx.x * blockDim.x + threadIdx.x;
-  auto const stride    = blockDim.x * gridDim.x;
+  auto const start_idx = cudf::detail::grid_1d::global_thread_id();
+  auto const stride    = cudf::detail::grid_1d::grid_stride();
   assert(start_idx < num_states);
 
   curandState localState = state[start_idx];
 
-  for (size_type idx = start_idx; idx < build_tbl_size; idx += stride) {
+  for (thread_index_type tidx = start_idx; tidx < build_tbl_size; tidx += stride) {
```

Comment:
Should we add a range-based for loop to replace this pattern?
```
for(auto const tidx: grid_1d::thread_indices()) {...}
```
So we won't need to have `stride` like this.

Reply:
For now I'm going to leave that out of scope for this PR. Please feel free to comment on https://github.com/rapidsai/cudf/issues/10368 -- the topic of a range-based loop has already come up there.